### PR TITLE
feat(mantine): mantine package creation

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Initial creation of the `@coveord/plasma-mantine` package. Took the code we have in the admin-ui and moved it here in this new package.

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/mantine/package.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
+{
+    "name": "@coveord/plasma-mantine",
+    "version": "0.0.0",
```
Comment (gdostie) [inline packages/mantine/package.json:3]: Not sure what version it should initially have 🤔

Comment (aiheon) [reply packages/mantine/package.json:3]: I would start it at v1.0.0 I think? It was working in the admin so it should be mature enough to be a major? 🤔

File: packages/mantine/package.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        "@swc/cli": "0.1.57",
+        "@swc/core": "1.3.3",
+        "@swc/jest": "0.2.22",
+        "@testing-library/dom": "8.13.0",
+        "@testing-library/jest-dom": "5.16.5",
+        "@testing-library/react": "11.2.7",
+        "@testing-library/react-hooks": "8.0.1",
+        "@testing-library/user-event": "12.8.3",
+        "@types/jest": "27.5.2",
+        "@types/lodash.defaultsdeep": "^4.6.7",
```
Comment (aiheon) [inline packages/mantine/package.json:60]:
```suggestion
        "@types/lodash.defaultsdeep": "4.6.7",
```

File: packages/mantine/package.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        "react-redux": "7.2.8",
+        "redux": "4.2.0"
+    },
+    "devDependencies": {
+        "@emotion/react": "^11.10.0",
+        "@mantine/core": "^5.6.2",
+        "@mantine/dates": "^5.6.2",
+        "@mantine/form": "^5.6.2",
+        "@mantine/hooks": "^5.6.2",
+        "@mantine/modals": "^5.6.2",
```
Comment (aiheon) [inline packages/mantine/package.json:50]:
```suggestion
        "@emotion/react": "11.10.0",
        "@mantine/core": "5.6.2",
        "@mantine/dates": "5.6.2",
        "@mantine/form": "5.6.2",
        "@mantine/hooks": "5.6.2",
        "@mantine/modals": "5.6.2",
```

File: packages/mantine/package.json (hunk 1 above)
Comment (GermainBergeron) [reply packages/mantine/package.json:3]: It will automatically bump to 47 or whatever when we merge the PR

**Review by GermainBergeron — APPROVED** [review]

File: packages/mantine/package.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        "react-redux": "7.2.8",
+        "redux": "4.2.0"
+    },
+    "devDependencies": {
+        "@emotion/react": "11.10.0",
+        "@mantine/core": "5.6.2",
+        "@mantine/dates": "5.6.2",
+        "@mantine/form": "5.6.2",
+        "@mantine/hooks": "5.6.2",
+        "@mantine/modals": "5.6.2",
```
Comment (toofff) [inline packages/mantine/package.json:50]: Shouldn't this be dependencies?

File: packages/mantine/package.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        "@tanstack/table-core": "8.5.11",
+        "dayjs": "1.11.3",
+        "lodash.defaultsdeep": "4.6.1",
+        "monaco-editor": "0.34.0",
+        "react-beautiful-dnd": "13.1.0",
+        "react-redux": "7.2.8",
+        "redux": "4.2.0"
+    },
+    "devDependencies": {
+        "@emotion/react": "11.10.0",
```
Comment (toofff) [inline packages/mantine/package.json:45]: Warning, this dependency is a duplicate

File: packages/mantine/package.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        "@types/react-redux": "7.1.24",
+        "@types/testing-library__jest-dom": "5.14.5",
+        "eslint-plugin-jest": "26.1.5",
+        "eslint-plugin-testing-library": "5.5.1",
+        "identity-obj-proxy": "3.0.0",
+        "jest": "28.1.2",
+        "jest-environment-jsdom": "28.1.2",
+        "jest-junit": "13.2.0",
+        "npm-run-all": "4.1.5",
+        "react": "17.0.2",
```
Comment (toofff) [inline packages/mantine/package.json:73]: Shouldn't this be dependency?

File: packages/mantine/package.json [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        "clean": "rimraf dist",
+        "start": "node ../../scripts/start",
+        "test": "jest --maxWorkers=65%"
+    },
+    "dependencies": {
+        "@coveord/plasma-react-icons": "workspace:*",
+        "@coveord/plasma-tokens": "workspace:*",
+        "@emotion/react": "11.10.0",
+        "@monaco-editor/react": "4.4.5",
+        "@swc/helpers": "0.4.11",
```
Comment (toofff) [inline packages/mantine/package.json:34]: Shouldn't this be dev dependency?

File: packages/mantine/package.json [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        "@tanstack/react-table": "8.5.11",
+        "@tanstack/table-core": "8.5.11",
+        "dayjs": "1.11.3",
+        "lodash.defaultsdeep": "4.6.1",
+        "monaco-editor": "0.34.0",
+        "react-beautiful-dnd": "13.1.0",
+        "react-redux": "7.2.8",
+        "redux": "4.2.0"
```
Comment (toofff) [inline packages/mantine/package.json:42]: Are these still valid dependencies?

File: packages/mantine/package.json [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        "@testing-library/jest-dom": "5.16.5",
+        "@testing-library/react": "11.2.7",
+        "@testing-library/react-hooks": "8.0.1",
+        "@testing-library/user-event": "12.8.3",
+        "@types/jest": "27.5.2",
+        "@types/lodash.defaultsdeep": "4.6.7",
+        "@types/react": "17.0.44",
+        "@types/react-beautiful-dnd": "13.1.2",
+        "@types/react-dom": "17.0.15",
+        "@types/react-redux": "7.1.24",
```
Comment (toofff) [inline packages/mantine/package.json:64]: Are these still valid dependencies?

File: packages/mantine/package.json (hunk 7 above)
Comment (GermainBergeron) [reply packages/mantine/package.json:34]: No, it's a runtime dependency

File: packages/mantine/package.json [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    "version": "0.0.0",
+    "description": "A Plasma flavoured Mantine theme",
+    "keywords": [
+        "plasma",
+        "mantine",
+        "components",
+        "coveo"
+    ],
+    "license": "ISC",
+    "author": "",
```
Comment (toofff) [inline packages/mantine/package.json:12]:
```suggestion
    "license": "Apache-2.0",
    "author": "Coveo",
```

File: packages/mantine/package.json (hunk 5 above)
Comment (gdostie) [reply packages/mantine/package.json:45]: How is it a duplicate?

File: packages/mantine/package.json (hunk 4 above)
Comment (gdostie) [reply packages/mantine/package.json:50]: We decided to put them in the peerDeps + devDeps to allow users of plasma-mantine to have more control over the version of mantine they want to use. Just like we do for react and react-dom

File: packages/mantine/package.json (hunk 9 above)
Comment (gdostie) [reply packages/mantine/package.json:64]: no good catch I will remove those!

File: packages/mantine/package.json (hunk 5 above)
Comment (gdostie) [reply packages/mantine/package.json:43]: oh nvm, it found it

**Review by toofff — APPROVED** [review]

**Review by aiheon — APPROVED** [review]

**Pull request merged**